### PR TITLE
Fix RootResults default for scipy dev version

### PR DESCRIPTION
**Pull request by QRemy**
Fix RootResults default for scipy dev version

### CODE REVIEW

File: gammapy/utils/roots.py [hunk 1]
```diff
@@ -6,7 +6,12 @@
 import astropy.units as u
 from gammapy.utils.interpolation import interpolation_scale
 
-BAD_RES = RootResults(root=np.nan, iterations=0, function_calls=0, flag=0)
+try:
```
Comment (registerrier) [inline gammapy/utils/roots.py:9]: Why is the try needed? Why does this yield a `TypeError` in some cases?

Comment (QRemy) [reply gammapy/utils/roots.py:9]: method is a new argument in the dev version, is there a better way to do this ?

**Comment by registerrier** [comment]
@MeeseeksDev backport to v1.0.x

**Comment by registerrier** [comment]
@meeseeksdev backport to v1.1.x

**Pull request merged**